components/client.jsx

```diff
@@ -26,14 +26,9 @@ function createElement(Component, props) {
 }
 
 function onEnter(nextState, replaceState) {
-  var pathname = langURLParser(nextState.location);
-  var query = nextState.location.query;
-  if (pathname) {
-    if (query.redirect) {
-      pathname += query.query || "";
-      query = {};
-    }
-    replaceState({}, pathname, query);
+  var result = redirectURLParser(nextState.location);
```

Comment:
I moved the above functionality into langURLParser to make the functionality easily testable.

I also changed the name of the function as it's doing more than just locale redirects, but generic redirects like `en-US/nseobius` and redirects with query unrelated to the locale.